src/registers/rflags.rs

```diff
@@ -108,7 +108,8 @@ mod x86_64 {
     pub fn write_raw(val: u64) {
         #[cfg(feature = "inline_asm")]
         unsafe {
-            llvm_asm!("pushq $0; popfq" :: "r"(val) : "memory" "flags")
+            // FIXME - There's probably a better way than saying we preserve the flags even though we actually don't
+            asm!("push {}; popf", in(reg) val, options(preserves_flags))
```

Comment:
Could you clarify your reason for setting the `preserves_flags` option here? Why not just omit it?

Reply:
The compiler restores CPU flags on exit of an assembly block unless you include that